app/redux/store.js

```diff
@@ -1,16 +1,86 @@
-import { applyMiddleware, compose, createStore, combineReducers } from 'redux'
-import logger from 'redux-logger'
-import thunk from 'redux-thunk'
-import * as reducers from './reducers/connection'
+import {applyMiddleware, compose, createStore, combineReducers} from 'redux';
+import logger from 'redux-logger';
+import thunk from 'redux-thunk';
+import * as reducers from './reducers/connection';
+import ZMQ from 'zmq';
+import MsgPack from 'msgpack';
 
-let createStoreWithMiddleware;
+const requester = ZMQ.socket('req');
+const subscriber = ZMQ.socket('sub');
 
-createStoreWithMiddleware = compose(
-  applyMiddleware(logger(), thunk)
-)(createStore)
+const zmqMiddlewareCreator = (zmq, subzmq) => {
+  return ({dispatch}) => {
+    return next => action => {
```

Comment:
AFAIK every middleware is curried function. and `applyMiddleware` accepts a series of such functions and maps through it.

I think a part of magic happens [here](https://github.com/reactjs/redux/blob/master/src/applyMiddleware.js#L29-L30).